warehouse/macaroons/security_policy.py

```diff
@@ -90,16 +89,18 @@ def unauthenticated_userid(self, request):
         # fetch the user that is associated with it.
         macaroon_service = request.find_service(IMacaroonService, context=None)
         userid = macaroon_service.find_userid(macaroon)
-        if userid is not None:
-            return str(userid)
+
+        # TODO: This will return either a user or a project soon.
+        login_service = request.find_service(IUserService, context=None)
+        return login_service.get_user(userid)
```

Comment:
`macaroon_service.find_userid` extracts the macaroon ID from the macaroon and then fetches the macaroon with the ORM joining with the user table, then extracts the user id from the user, and discards both the macaroon instance and the user instance. And then you make a second request to get the user from its id.

You'd gain a DB request by doing 
```python
        user = macaroon_service.find_from_raw(macaroon).user
```

(also, I wonder why there's no exception handling here but there wasn't before either)

Reply:
This isn't going to be around very long, as we'll need a way to get either the User or the Project from the macaroon, but I'd be fine with making this change for now to make this slightly more efficient.

Reply:
5cdb53a6